spaghetto/src/Spago/Command/Ls.purs

```diff
@@ -1,3 +1,120 @@
-module Spago.Command.Ls where
+module Spago.Command.Ls (listPackages, listPackageSet, LsEnv(..), JsonFlag(..)) where
 
-import Prelude
+import Spago.Prelude
+
+import Data.Array (replicate)
+import Data.Codec.Argonaut as CA
+import Data.Codec.Argonaut.Record as CAR
+import Data.Foldable (elem, maximum)
+import Data.Map (filterKeys)
+import Data.Map as Map
+import Data.Set (unions)
+import Data.String.CodeUnits (fromCharArray, length)
+import Data.Tuple.Nested (type (/\))
+import Registry.PackageName as PackageName
+import Registry.Version as Version
+import Spago.Bin.Flags (IncludeTransitive(..))
+import Spago.Config (Dependencies(..), Package(..), PackageSet, Workspace, getPackageLocation)
+import Spago.Config as Config
+import Spago.Json (stringifyJson)
+
+data JsonFlag = JsonOutputNo | JsonOutputYes
+
+type LsEnv =
+  { dependencies :: PackageSet
+  , logOptions :: LogOptions
+  , workspace :: Workspace
+  }
+
+type JsonPackageOutput =
+  { packageName :: String
+  -- TODO: Was Json Value, has tag
+  -- See https://github.com/purescript/spago/blob/ec22308d181340382494e5bd6ef9102a3c87c0bb/src/Spago/Types.hs#L38-L47
+  , repo :: String
+  , version :: String
+  }
+
+jsonPackageOutputCodec :: JsonCodec JsonPackageOutput
+jsonPackageOutputCodec = CAR.object "JsonPackageOutput"
+  { packageName: CA.string
+  , repo: CA.string
+  , version: CA.string
+  }
+
+listPackageSet :: JsonFlag -> Spago LsEnv Unit
+listPackageSet jsonFlag = do
+  logDebug "Running `listPackageSet`"
+  { workspace } <- ask
+  output $ formatPackageNames jsonFlag (Map.toUnfoldable workspace.packageSet)
+
+listPackages :: IncludeTransitive -> JsonFlag -> Spago LsEnv Unit
```

Comment:
Let's pass in the json flag as a boolean too. In general we pass a record of options to commands, so you can squash these parameters in a single record.

Reply:
Done in 50d81ddbfb53a39e2f3914c98ea6c480adce2578